src/modules/Elsa.Workflows.Core/Contexts/WorkflowExecutionContext.cs

```diff
@@ -543,16 +419,14 @@ public T UpdateProperty<T>(string key, Func<T?, T> updater)
         return value;
     }
 
-    /// <summary>
     /// Returns true if the <see cref="Properties"/> dictionary contains the specified key.
-    /// </summary>
     public bool HasProperty(string name) => Properties.ContainsKey(name);
 
-    internal bool CanTransitionTo(WorkflowSubStatus targetSubStatus) => ValidateStatusTransition(targetSubStatus);
+    internal bool CanTransitionTo(WorkflowSubStatus targetSubStatus) => ValidateStatusTransition();
 
     internal void TransitionTo(WorkflowSubStatus subStatus)
     {
-        if (!ValidateStatusTransition(SubStatus))
+        if (!ValidateStatusTransition())
```

Comment:
```suggestion
        if (!ValidateStatusTransition(subStatus))
```

Reply:
```suggestion
// TODO: Check if this method should be using `targetSubStatus` instead of `SubStatus`.
private bool ValidateStatusTransition(WorkflowSubStatus targetSubStatus)
    private bool ValidateStatusTransition()
    {
        var currentMainStatus = GetMainStatus(SubStatus);
```